HeadwayTests/HeadwayTests.swift

```diff
@@ -1,17 +1,339 @@
+        state.duration = sampleBook.chapters[0].duration
+        state.playbackProgress = 0.5
+        state.isPlaying = true
+
+        let store = await TestStore(initialState: state) {
+            BookPlayerFeature()
+        } withDependencies: {
+            $0.playerClient.play = { _, _ in }
+        }
 
-    @Test func example() async throws {
-        // Write your test here and use APIs like `#expect(...)` to check expected conditions.
+        await store.send(.selectNexChapter) {
+            $0.selectedChapterIndex = 1
+            $0.duration = sampleBook.chapters[1].duration
+            $0.playbackProgress = 0.0
+        }
     }
 
+    @Test
+    func seekSuccess() async {
+        var state = BookPlayerFeature.State()
+        state.playbackProgress = 0.3
+
+        let store = await TestStore(initialState: state) {
+            BookPlayerFeature()
+        } withDependencies: {
+            $0.playerClient.seek = { _ in }
+        }
+
+        await store.send(.seek(0.5)) {
+            $0.playbackProgress = 0.5
+        }
+    }
+
+    @Test
+    func seekFailure() async {
+        var state = BookPlayerFeature.State()
+        state.playbackProgress = 0.3
+        let error = NSError(domain: "testing", code: 999)
+
+        let store = await TestStore(initialState: state) {
+            BookPlayerFeature()
+        } withDependencies: {
+            $0.playerClient.seek = { _ in
+                throw error
+            }
+        }
+
+        await store.send(.seek(0.5)) {
+            $0.playbackProgress = 0.5
+        }
+
+        await store.receive(.playerErrored(error.localizedDescription)) {
+            $0.alert = AlertState {
+                TextState("Something went wrong!")
+            } actions: {
+                ButtonState(role: .cancel, action: .dismiss) {
+                    TextState("OK")
+                }
+            } message: {
+                TextState(error.localizedDescription)
+            }
+        }
+    }
+
+    @Test
+    func updateProgressUpdatesState() async {
+        var state = BookPlayerFeature.State()
+        state.duration = 300
+        state.playbackProgress = 0.0
+
+        let store = await TestStore(initialState: state) {
+            BookPlayerFeature()
+        }
+
+        await store.send(.updateProgress(150)) {
+            $0.playbackProgress = 150 / 300.0
+        }
+    }
+
+    @Test
+    func updateProgressTriggersChapterFinished() async {
+        var state = BookPlayerFeature.State()
+        state.book = sampleBook
+        state.duration = 300
+        state.playbackProgress = 0.0
+
+        let store = await TestStore(initialState: state) {
+            BookPlayerFeature()
+        }
+
+        await store.send(.updateProgress(299))
+        await store.receive(.chapterFinished) {
+            $0.selectedChapterIndex = 1
+            $0.duration = sampleBook.chapters[1].duration
+            $0.playbackProgress = 0.0
+        }
```

Comment:
Because of the approach based on `tolerance` this test will succeed even on a 297's second, and the next chapter will be selected. Not ideal, but managable.